corpus/expressions.txt

```diff
@@ -1086,3 +1086,21 @@ yield db.users.where('[endpoint+email]')
           (member_expression (identifier) (property_identifier))
           (property_identifier))
         (arguments (string))))))
+
+============================================
+JSX
+============================================
+
+var a = <Foo></Foo>
+b = <Foo.Bar></Foo.Bar>
+c = <> <Foo /> </>
+d = <Bar> <Foo /> </Bar>
+e = <Foo bar/>
+f = <Foo bar="string" baz={2} data-i8n="dialogs.welcome.heading" bam />
+g = <Avatar userId={foo.creatorId} />
+h = <input checked={this.state.selectedNewStreetType === 'new-street-default' || !this.state.selectedNewStreetType}> </input>
+i = <Foo:Bar bar={}>{...children}</Foo:Bar>
+
+---
+
+(program (variable_declaration (variable_declarator (identifier) (jsx_element (jsx_opening_element (identifier)) (jsx_closing_element (identifier))))) (expression_statement (assignment_expression (identifier) (jsx_element (jsx_opening_element (jsx_member_expression (identifier) (identifier))) (jsx_closing_element (jsx_member_expression (identifier) (identifier)))))) (expression_statement (assignment_expression (identifier) (jsx_fragment (jsx_text) (jsx_self_closing_element (identifier)) (jsx_text)))) (expression_statement (assignment_expression (identifier) (jsx_element (jsx_opening_element (identifier)) (jsx_text) (jsx_self_closing_element (identifier)) (jsx_text) (jsx_closing_element (identifier))))) (expression_statement (assignment_expression (identifier) (jsx_self_closing_element (identifier) (jsx_attribute (identifier))))) (expression_statement (assignment_expression (identifier) (jsx_self_closing_element (identifier) (jsx_attribute (identifier) (string)) (jsx_attribute (identifier) (jsx_expression (number))) (jsx_attribute (jsx_identifier (identifier) (identifier)) (string)) (jsx_attribute (identifier))))) (expression_statement (assignment_expression (identifier) (jsx_self_closing_element (identifier) (jsx_attribute (identifier) (jsx_expression (member_expression (identifier) (property_identifier))))))) (expression_statement (assignment_expression (identifier) (jsx_element (jsx_opening_element (identifier) (jsx_attribute (identifier) (jsx_expression (binary_expression (binary_expression (member_expression (member_expression (this) (property_identifier)) (property_identifier)) (string)) (unary_expression (member_expression (member_expression (this) (property_identifier)) (property_identifier))))))) (jsx_text) (jsx_closing_element (identifier))))) (expression_statement (assignment_expression (identifier) (jsx_element (jsx_opening_element (jsx_namespace_name (identifier) (identifier)) (jsx_attribute (identifier) (jsx_expression))) (jsx_expression (spread_element (identifier))) (jsx_closing_element (jsx_namespace_name (identifier) (identifier)))))))
```

Comment:
This test is kinda hard to read w/ the single line. Can you format it a bit? We should really add some automated way to do it, I know it can be a bit tedious.